lib/charms/hydra/v0/oauth.py

```diff
@@ -373,7 +376,12 @@ def get_provider_info(self, relation_id: Optional[int] = None) -> Optional[Dict]
         if not relation:
```

Comment:
Since you are fixing bugs, lets check that `app` also exists
```suggestion
        if not relation or not relation.app:
```